br/pkg/storage/local.go

```diff
@@ -28,12 +31,30 @@ const (
 // export for using in tests.
 type LocalStorage struct {
 	base string
+	// Whether ignoring ENOINT while deleting.
+	// Don't fail when deleting an unexist file is more like
+	// a normal ExternalStorage implementation does.
+	IgnoreEnoentForDelete bool
+}
+
+// Base returns the base dir used by this local storage.
+func (l *LocalStorage) Base() string {
+	return l.base
 }
 
 // DeleteFile deletes the file.
 func (l *LocalStorage) DeleteFile(_ context.Context, name string) error {
+	failpoint.Inject("local_delete_file_err", func(v failpoint.Value) {
+		failpoint.Return(errors.New(v.(string)))
+	})
 	path := filepath.Join(l.base, name)
-	return os.Remove(path)
+	err := os.Remove(path)
+	if err != nil &&
+		l.IgnoreEnoentForDelete &&
+		stderrors.Is(err, syscall.ENOENT) {
```

Comment:
local external storage also support windows, so can `syscall.ENOENT` also match the error occur in windows system?

Reply:
I guess for now TiDB doesn't have any compatibility promise for windows?

Reply:
I have checked that this won't make TiDB failed to compile in Windows. As this only affects test cases, I think it could be fine to don't specially handle this in Windows system.